### PR TITLE
QRCode close on  onClose

### PR DESCRIPTION
**Pull request by BriceShatzer**
- Actually close the QRCode component on the onClose call per: 
https://engineering.paypalcorp.com/jira/browse/DTCHKINT-925 & https://engineering.paypalcorp.com/jira/browse/DTCHKINT-887

- Wires onCancel callback to use the error UI state per:
 https://engineering.paypalcorp.com/jira/browse/DTCHKINT-893

- Removes `setProps` call that was used early on in QA & testing
https://engineering.paypalcorp.com/jira/browse/DTCHKINT-886

### CODE REVIEW

**Review by nbierdeman — COMMENTED** [review]
@BriceShatzer, I rebased this branch with `master` locally, and all of the tests passed. 👍

File: src/payment-flows/native/native.js [hunk 1]
```diff
@@ -306,23 +306,21 @@ function initNative({ props, components, config, payment, serviceData } : InitOp
...
+        const logQRClose = (event? : string = 'closeQRCode') => {
             getLogger().info(`VenmoDesktopPay_qrcode_closing_${ event }`).track({
                 [FPTI_KEY.STATE]:       FPTI_STATE.BUTTON,
                 [FPTI_KEY.TRANSITION]:  event ? `${ FPTI_TRANSITION.QR_CLOSING }_${ event }` : FPTI_TRANSITION.QR_CLOSING
             }).flush();
-            return onCloseCallback();
         };
         
-        return orderIDPromise.then((orderID) => {
+        orderIDPromise.then((orderID) => {
```
Comment (gregjopa) [inline src/payment-flows/native/native.js:316]: Could we add this `return` back? Or is returning a promise in `initQRCode()` messing with the close functionality?

**Review by nbierdeman — APPROVED** [review]
🥳

**Review by remotevision — APPROVED** [review]
lgtm!

**Review by gregjopa — APPROVED** [review]
Looks good!

**Pull request merged**